testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1,7 +1,12 @@
 use std::io::Cursor;
+use async_std::io::ReadExt;
 use std::time::Instant;
 
-use reqwest::blocking::{Client, RequestBuilder};
+use async_h1::{client};
```

Comment:
If the only thing we need `async-h1` for is the client, then we might as well continue to use `reqwest`.

Reply:
This is not the case, we're using async-h1 for its server as well. Out of curiosity, why would keep reqwest instead? It contains unsafe code, and pulls way more dependencies.

Reply:
I'm mainly concerned by the fact that we're now using _two_ HTTP client libraries. 
 By the time we release `mainnet`, we will have replaced whatever 3rd party library with our own code, so all the `unsafe` code will not be part of the build by then anyway.  But, to make it easier to replace whatever library we use, we should stick to just one HTTP client (and avoid using `async/await` if possible).  I'm fine with using `async-h1` over `reqwest`, but if so, then we should drop `reqwest` altogether.

Reply:
> we're now using two HTTP client libraries

One of them is a dev dependency, though. And I think should that these HTTP libraries (WARP and reqwest) should always remain a dev dependency -- the tests should not be rewritten to use our HTTP code, as using a convenient second HTTP implementation for tests helps insure that our HTTP code is compliant.

Reply:
The goal of the refactoring included in this PR is specifically addressing this: use **one**, safe, lighter, HTTP client library - `async-h1`.
Are you concerned because we're still using different libraries (`reqwest`, `tokio` and `warp`) for making the process of writing tests more convenient?

Reply:
Ah, I see that now.  When I wrote this, I had thought that `reqwest` was still being used outside of tests.

Reply:
The use of `async-h1` to replace `reqwest` in all non-test use-cases is fine with me.